doc/productivity.md

```diff
@@ -172,11 +172,35 @@ When looking at other's pull requests, it may make sense to add the following se
 
 ```
 [remote "upstream-pull"]
-        fetch = +refs/pull/*:refs/remotes/upstream-pull/*
+        fetch = +refs/pull/*/head:refs/remotes/upstream-pull/*
         url = git@github.com:bitcoin/bitcoin.git
 ```
 
-This will add an `upstream-pull` remote to your git repository, which can be fetched using `git fetch --all` or `git fetch upstream-pull`. Afterwards, you can use `upstream-pull/NUMBER/head` in arguments to `git show`, `git checkout` and anywhere a commit id would be acceptable to see the changes from pull request NUMBER.
+This will add an `upstream-pull` remote to your git repository, which can be fetched using `git fetch --all` or `git fetch upstream-pull`. It will download and store on disk about 1 GB of data (all PRs, including merged and closed ones). Afterwards, you can use `upstream-pull/NUMBER/head` in arguments to `git show`, `git checkout` and anywhere a commit id would be acceptable to see the changes from pull request NUMBER.
+
+It is also possible to fetch only a given pull request by doing
+
+```
+git fetch origin pull/NUMBER/head:LOCAL_BRANCH_NAME_TO_BE_CREATED
+```
+
+for example:
+
+```
+git fetch origin pull/18391/head:docs_pr
+```
+
+or
+
+```
+git fetch https://github.com/SUBMITTER/bitcoin/ PR_BRANCH:LOCAL_BRANCH_NAME_TO_BE_CREATED
+```
+
+for example:
+
+```
+git fetch https://github.com/glowang/bitcoin/ update_blocksonly_docs:docs_pr
```

Comment:
Have you checked with this person about using their identity here?

Reply:
Ping @glowang.

Reply:
thank you for asking - this is fine with me!